App/Models/TextileTypes.ts

```diff
@@ -1,19 +1,42 @@
+// BEGIN API Types
+
+// private enums to enforce strict id types
+// read here: https://basarat.gitbooks.io/typescript/docs/tips/nominalTyping.html#using-enums
+enum TextileIdBase {}
+enum BlockIdBase {}
+enum ThreadIdBase {}
+enum PhotoIdBase {}
+enum PeerIdBase {}
+enum DeviceIdBase {}
+enum ProfileAvatarIdBase {}
+enum NotificationIdBase {}
+export type TextileId = TextileIdBase & string & undefined
+export type BlockId = BlockIdBase & string & undefined
+export type ThreadId = ThreadIdBase & string & undefined
+export type PhotoId = PhotoIdBase & string & undefined
+export type PeerId = PeerIdBase & string & undefined
+export type DeviceId = DeviceIdBase & string & undefined
+export type ProfileAvatarId = ProfileAvatarIdBase & string & undefined
+export type NotificationId = NotificationIdBase & string & undefined
```

Comment:
the enum base makes it so that these aren't simply strings that can be treated as strings, you'll actually get back an error if you try to pass, for example, a PeerId as a BlockId

Reply:
the undefined option is common for base types but here isn't, so when creating composite types that are like `preferences = {photoId?: TextileType.PhotoId}` you can't set it since PhotoId type is never undefined, but the photoId is looking for an either/or.